docs/rule_authors/writing_toolchains.md

```diff
@@ -0,0 +1,130 @@
+---
+id: writing_toolchains
+title: Writing Toolchains
+---
+
+Toolchains are regular rules that:
+
+- Have `is_toolchain_rule = True` passed to the `rule` call.
```

Comment:
What does this actually do? I haven't been able to find any discussion.

Reply:
[I don't know either](https://github.com/facebook/buck2/issues/666)! I haven't gotten any answers from Meta, but no luck so far.
I've seen that it's actually used in the Rust code, but haven't taken the time to dig deeper.

Reply:
I noticed that using a toolchain target as a non-toolchain dependency produces an error. Perhaps that's the sum total effect?

Reply:
I don't think there's any action item here. Can we resolve this?
I suggest continuing the conversation somewhere else if you're still interested.

Reply:
I think this would be a great place to document what this option does, but I guess it can be left for future work.